### PR TITLE
[Azure Search] Generate code for the Index and Suggest APIs

### PR DESCRIPTION
**Pull request by bitsforthought**
Corresponding Swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/5219

This change replaces a bunch of custom code with generated code, specifically for the Index and Suggest APIs.

There are some breaking changes:
  1. DocumentSuggestResult/Base and SuggestResult/Base have been removed. For dynamic typing scenarios, use DocumentSuggestResult<Document> and SuggestResult<Document>.
  2. IndexBatch/Base and IndexAction/Base have been removed. For dynamic typing scenarios, use IndexBatch<Document> and IndexAction<Document>.
We have already bumped the major SDK version in anticipation of these breaking changes.

Big thanks to @mhko who did the majority of the work on this.

FYI @arv100kri @Yahnoosh

### CODE REVIEW

**Comment by arv100kri** [comment]
Is it by design that the Generated metadata points to your fork and not the public Azure repo?
Is that why [DO NOT MERGE] ?

**Comment by brjohnstmsft** [comment]
@arv100kri Yes, thanks for reminding me. I still need to update that. Will push another commit in a little while.

Also, the [DO NOT MERGE] is to let azure-sdk-for-net maintainers know that we (Azure Search team) will take care of merging this, since it targets the search-preview branch.

**Comment by arv100kri** [comment]
>             }

Pretty cool that the new generated client/proxy operations have this check :)

---
Refers to: src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Data/Customizations/Documents/DocumentsOperations.Customization.cs:240 in 7c11fd1. [](commit_id = 7c11fd1e5538a38a0c286791c82f245997c68b54, deletion_comment = True)

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Data/Customizations/Documents/DocumentsOperations.Customization.cs
Comment (arv100kri) [inline src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Data/Customizations/Documents/DocumentsOperations.Customization.cs]: Pretty cool that the new generated client/proxy operations have this check :)

**Review by arv100kri — APPROVED** [review]

**Comment by natinimni** [comment]
LGTM

**Comment by UD1412** [comment]
sign-off

**Pull request merged**
